networkx/generators/random_graphs.py

```diff
@@ -152,10 +162,11 @@ def gnp_random_graph(n, p, seed=None, directed=False):
     """
     if directed:
         edges = itertools.permutations(range(n), 2)
-        G = nx.DiGraph()
+        G = (create_using or nx.DiGraph)()
```

Comment:
The difficulty here is that `create_using` is usually a type but it can also be a graph instance -- in which case we clear it before construction. This is usually done by using 
`nx.empty_graph(N, create_using=create_using)` and if you want to set the default use
`nx.empty_graph(N, create_using=create_using, default=nx.DiGraph if directed else nx.Graph)` or similar.
```suggestion
        G = nx.empty_graph(n, create_using, nx.DiGraph if directed else nx.Graph)
```
The constructors like `nx.complete_graph` do that for you so `nx.complete_graph(n, create_using=create_using)` should be fine.